include/tables/XAttrTable.h

```diff
@@ -163,7 +77,7 @@ class XAttrTable : public DBTable<XAttrRow> {
 
 public:
 
-  XAttrTable() : DBTable("hdfs_xattrs"){
+  XAttrTable(string table_name) : DBTable(table_name){
```

Comment:
You should keep the old constructor as it is and add this new constructor as protected. We shouldn't expose the table name as parameter outside the class .